lib/semantics/check-omp-structure.cc

```diff
@@ -177,7 +201,31 @@ void OmpStructureChecker::Leave(const parser::OmpClauseList &) {
 
       // TODO: ordered region binding check (requires nesting implementation)
     }
-  }
+  }  // DO
+
+  // 2.8.1 Simd Construct Restriction
+  if (GetContext().directive == OmpDirective::SIMD) {
+    if (auto *clause{FindClause(OmpClause::SIMDLEN)}) {
```

Comment:
Here you are checking to see if you have two clauses and fetching their integer values.  Is this going to be a common pattern?  If so, you might want to create function that does the same & returns a tuple of two optionals (or whatever).

Reply:
> Here you are checking to see if you have two clauses and fetching their integer values. Is this going to be a common pattern? 
For this pattern, I think we've dealt with all the cases based on the spec.